File: solutions/julia/bird-watcher/1/bird-watcher.jl
function today(birds_per_day)
    return birds_per_day[end]
end

function increment_todays_count(birds_per_day)
    birds_per_day[end] += 1
    return birds_per_day
end

function has_day_without_birds(birds_per_day)
    return !(isempty(birds_per_day[iszero.(birds_per_day)]))
end

function count_for_first_days(birds_per_day, num_days)
    return sum(birds_per_day[begin:num_days])
end

function busy_days(birds_per_day) 
    return length(birds_per_day[birds_per_day .>= 5])
end

function average_per_day(week1, week2)
    CONST = 2
    sum = week1 .+ week2
    mean = sum ./ CONST
    return mean
end

File: solutions/r/bird-watcher/1/bird-watcher.R
today <- function(birds_per_day) {
  tail(birds_per_day, 1)
}

increment_todays_count <- function(birds_per_day) {
  birds_per_day[length(birds_per_day)] <- birds_per_day[length(birds_per_day)] + 1
  birds_per_day
}

has_day_without_birds <- function(birds_per_day) {
  any(birds_per_day == 0)
}

count_for_first_days <- function(birds_per_day, num_days) {
  sum(birds_per_day[1:num_days])
}

busy_days <- function(birds_per_day) {
  length(birds_per_day[birds_per_day >= 5])
}

running_total <- function(birds_per_day) {
  cumsum(birds_per_day)
}

busy_days_of_week <- function(birds_per_day, day_names) {
  day_names[order(birds_per_day, decreasing=T)]
}
